src/lib/components/composites/paper-components/PaperEntry.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+    import PaperInfo from "$lib/components/composites/paper-components/PaperInfo.svelte";
+    import { type Paper, ReviewDecision, ReviewProgress, type ReviewStatus } from "$lib/types";
+    import UserAvatar from "$lib/components/composites/user-avatar/UserAvatar.svelte";
+    import { goto } from "$app/navigation";
+
+    interface PaperEntryProps {
+        paper: Paper;
+        projectId: number;
+        showReviewStatus?: boolean;
+        onClick?: () => void;
+    }
+
+    const navigateToPaperView = () => goto(`/project/${projectId}/paper/${paper.id}`);
+
+    const {
+        paper,
+        projectId,
+        showReviewStatus = false,
+        onClick = navigateToPaperView,
+    }: PaperEntryProps = $props();
+
+    // Mapping of review status to border color of paper entry
+    const reviewStatusColor: Record<ReviewStatus, string> = {
+        [ReviewProgress.Unreviewed]: "border-unreviewed-gray",
+        [ReviewDecision.Accepted]: "border-accept-green",
+        [ReviewDecision.Maybe]: "border-maybe-yellow",
+        [ReviewDecision.Declined]: "border-decline-red",
+    };
+
+    const showStatus = () => showReviewStatus && paper.reviewData !== undefined;
+
+    let timeoutId: ReturnType<typeof setTimeout> | null = null;
+    /**
+     * Handles the click event of the paper entry component by checking
+     * whether it was a single click (so no further click after 250ms) or a double click
+     * and call the corresponding functions:
+     *  - single click => onClick() (possible overridden, otherwise navigateToPaperView())
+     *  - double click => navigateToPaperView() (default)
+     *
+     * @param e - the click event
+     */
+    const handleClick = (e: MouseEvent) => {
+        if (e.detail === 1) {
+            if (timeoutId !== null) clearTimeout(timeoutId);
+            timeoutId = setTimeout(() => onClick(), 250);
+        } else if (e.detail === 2) {
+            if (timeoutId !== null) clearTimeout(timeoutId);
+            navigateToPaperView();
+        }
+    };
```

Comment:
This method looks a bit confusing at first. I think we can make it more self-describing.
```suggestion
    const handleClick = (event: MouseEvent) => {
        if (timeoutId !== null) {
            clearTimeout(timeoutId);
        }
        const numberOfClicks = event.detail;

        if (numberOfClicks === 1) {
            timeoutId = setTimeout(onClick, 250);
        } else if (numberOfClicks === 2) {
            navigateToPaperView();
        }
    };
```
- renamed `e` to `event`
- moved `timeoutId != null` check to the beginning
  - check is made for single and double click
  - else case isn't reached when we navigate to the paper view after a double click
- introduced new variable `numberOfClick` to make the meaning of `event.detail` more clear
- `() => onClick()` can be just `onClick` (the former just wraps `onClick` in another method)

What are your thoughts on these changes?

Reply:
I think your code is more elegant and I would change the timeout to 500ms, as you proposed (and Microsoft uses).
250ms was only a first try to make it faster, but I agree that the interval between the two clicks for a double click could be too short.

Reply:
I took a second look at this today and found some weird race conditions. If you click a second time, just when the timer stops, both events are fired. This is easy to achieve with a high cooldown, like 1 second.

I think this is due to the different interpretation of `numberOfClicks` and that the `timeoutId` isn't reset.

I investigated this for a while and found this, to be a sound solution:
- if the timeout wasn't set yet, it's the first click
  - start the timeout
  - when the timeout finishes, the `timeoutId` will be reset and the single click event fired
- when the timeout was set, it's the second click
  - clear the running timeout
  - reset `timoutId`
  - fire double click event
```ts
    const handleClick = () => {
        if (timeoutId === null) {
            timeoutId = setTimeout(() => {
                timeoutId = null;
                onClick();
            }, 350);
        } else {
            clearTimeout(timeoutId);
            timeoutId = null;
            navigateToPaperView();
        }
    };
```

What are your thoughts on this? Could you reproduce the behavior, which I described?

Also, on a second thought, 500 ms is too much, the single click has a high delay. We could do maybe 350ms?

Lastly, don't forget to adjust the documentation, there's a different amount of delay in it.

Reply:
I am also fine with 350ms, I also experimented with the delay an 250ms was the best for me, but I understand that it could be too fast for some people and we should increase it. Thus, i will change it (and update the documentation ;))